js/src/hooks/useUpsertAdsAccount.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useCallback } from '@wordpress/element'; // Add this line
+
+/**
+ * Internal dependencies
+ */
+import { useAppDispatch } from '.~/data';
+import { API_NAMESPACE } from '.~/data/constants';
+import useGoogleAdsAccount from './useGoogleAdsAccount';
+import useApiFetchCallback from './useApiFetchCallback';
+import useDispatchCoreNotices from '.~/hooks/useDispatchCoreNotices';
+
+/**
+ * Set up a Google Ads account.
+ *
+ * @return {[Function, Object]} Returns a function to create or update a Google Ads account and the data returned by the API request.
+ */
+const useUpsertAdsAccount = () => {
+	// Check if there is a connected Google Ads account which in this case will update the account.
+	// If not, it means we are creating a new account.
+	const { googleAdsAccount } = useGoogleAdsAccount();
+	const { createNotice } = useDispatchCoreNotices();
+	const { fetchGoogleAdsAccount, fetchGoogleAdsAccountStatus } =
+		useAppDispatch();
+
+	const [ fetchCreateAccount, data ] = useApiFetchCallback( {
+		path: `${ API_NAMESPACE }/ads/accounts`,
+		method: 'POST',
+		data: {
+			id: googleAdsAccount?.id || undefined,
+		},
+	} );
+
+	const upsertAdsAccount = useCallback( async () => {
+		try {
+			await fetchCreateAccount( { parse: false } );
+		} catch ( e ) {
+			// For status code 428, we want to allow users to continue and proceed,
+			// so we swallow the error for status code 428,
+			// and only display error message and exit this function for non-428 error.
+			if ( e.status !== 428 ) {
+				createNotice(
+					'error',
+					__(
+						'Unable to create Google Ads account. Please try again later.',
+						'google-listings-and-ads'
+					)
+				);
+			}
+		}
+
+		// Update Google Ads data in the data store after posting an account update.
+		await fetchGoogleAdsAccount();
+		await fetchGoogleAdsAccountStatus();
+	}, [
+		createNotice,
+		fetchCreateAccount,
+		fetchGoogleAdsAccount,
+		fetchGoogleAdsAccountStatus,
+	] );
+
+	return [ upsertAdsAccount, data ];
```

Comment:
https://github.com/woocommerce/google-listings-and-ads/blob/5c12ab7ee320521956a0e00f40867edad87f4acf/js/src/components/google-ads-account-card/create-account.js#L33-L34

Referring to the above use of the `loading` state in the `data`, the state is only related to `fetchCreateAccount` but doesn't cover the entire loading state of `upsertAdsAccount`.

I suppose that's one of the reasons the following derived state is incorrect.

https://github.com/woocommerce/google-listings-and-ads/blob/5c12ab7ee320521956a0e00f40867edad87f4acf/js/src/components/google-ads-account-card/create-account.js#L48-L50

Reply:
Responded in https://github.com/woocommerce/google-listings-and-ads/pull/2270#discussion_r1568015899